docker/build/x86_64/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# this dockerfile must be built from the top-level stash directory
+# ie from top=level stash:
+# docker build -t stash/build -f docker/build/x86_64/Dockerfile .
+
+FROM golang:1.11.13 as compiler
+
+RUN apt-get update && apt-get install -y apt-transport-https
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+
+RUN apt-get update && \
+    apt-get install -y nodejs yarn xz-utils --no-install-recommends || exit 1; \
+	rm -rf /var/lib/apt/lists/*;
+	
+ENV PACKR2_VERSION=2.0.2
+ENV PACKR2_SHA=f95ff4c96d7a28813220df030ad91700b8464fe292ab3e1dc9582305c2a338d2
+ENV PACKR2_DOWNLOAD_FILE=packr_${PACKR2_VERSION}_linux_amd64.tar.gz
+ENV PACKR2_DOWNLOAD_URL=https://github.com/gobuffalo/packr/releases/download/v${PACKR2_VERSION}/${PACKR2_DOWNLOAD_FILE}
+
+WORKDIR /
+RUN wget ${PACKR2_DOWNLOAD_URL}; \
+	echo "$PACKR2_SHA $PACKR2_DOWNLOAD_FILE" | sha256sum -c - || exit 1; \
+	tar -xzf $PACKR2_DOWNLOAD_FILE -C /usr/bin/ packr2; \
+	rm $PACKR2_DOWNLOAD_FILE;
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+RUN wget -O /ffmpeg.tar.xz https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.tar.xz && \
+    tar xf /ffmpeg.tar.xz && \
+    rm ffmpeg.tar.xz && \
+    mv /ffmpeg*/ /ffmpeg/
+
+# copy the ui yarn stuff so that it doesn't get rebuilt every time
+COPY ./ui/v2/package.json ./ui/v2/yarn.lock /stash/ui/v2/
+
+WORKDIR /stash
+RUN yarn --cwd ui/v2 install --frozen-lockfile
+
+COPY . /stash/
+ENV GO111MODULE=on
+
+RUN make generate 
+RUN make ui 
+RUN make build
+
+FROM ubuntu:19.10 as app
+
+RUN apt-get update && \
+    apt-get -y install ca-certificates && \
+    adduser stash --gecos GECOS --shell /bin/bash --disabled-password --home /home/stash
+COPY --from=compiler /stash/stash /ffmpeg/ffmpeg /ffmpeg/ffprobe /usr/bin/
+
```

Comment:
There should be a `USER` directive here declaring that the app itself should execute as the unprivileged user otherwise we just create the user for no reason.

The only issue this creates is file permissions corrections so if we can live with support requests for that its good to add the USER directive.

Reply:
This is copied from the production dockerfile, so is an existing issue. During testing, adding the `USER` directive resulted in a lot of permissions issues that I was unable to workaround. 

Having done some reading on this, this looks like a non-trivial issue that I'd prefer to push out to another issue and PR. 

[This stackoverflow answer](https://stackoverflow.com/a/39398511) suggests one solution using `ENTRYPOINT` to run the container as root, set the file permissions, then run stash as the stash user. Changing this around will also impact the `docker-compose` file.

Alternatively, I can remove the `adduser` from all of the docker files, but I'd rather keep it in there and fix the container user issue separately.

Reply:
I added it long before it was looking like anyone was really going to keep up on maintaining this and haven't really touched it since.  I'd suggest just remove it, for now, we can re-add it later.  

The only other thing we could do is have a tiny fix permissions go binary that would execute as you said a file permissions change on a static set of paths such that nobody can hijack or escalate anything. 

This could be run on container start with a `start.sh` or the `entrypoint.sh` really either is fine.   The biggest concern that I have is just anything that is done by the restricted user should be just that restricted.  

```c
#include <unistd.h>
int main() {
  // This program needs to run with setuid == root
  // This needs to be in a compiled language because you cannot setuid bash scripts
  setuid(0);
  execle("/bin/bash", "bash", "-c",
         "/bin/chown -R stash:stash /data && /bin/chmod -R 755 /data/",
         (char*) NULL, (char*) NULL);
}
```
This above little c code was used on another container I had worked on before to let the user summon the `chmod -R 755` via root level permissions except as the unpriv'd user.  I think that this method would be potentially ideal.  However it would likely be smarter if it handled a number of directories common to Docker but would be written in Go and be stored in another repository.

Reply:
How it was implemented looked like https://github.com/lbryio/lbry-docker/blob/fced92808905e6c23110d01c63df38d07290d5e2/lbrycrd/Dockerfile-linux-x86_64-production#L16